vision/api/label/snippets.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python
+
+# Copyright 2016 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import argparse
+import base64
+import httplib2
+import json
+
+from googleapiclient import discovery
+from oauth2client.client import GoogleCredentials
+
+DISCOVERY_URL = ('https://vision.googleapis.com/$discovery/rest?'
+                 'labels=TRUSTED_TESTER&version=v1')
+
+
+def get_service():
+    """Get vision service using discovery."""
+    credentials = GoogleCredentials.get_application_default()
+    scoped_credentials = credentials.create_scoped(
+        ['https://www.googleapis.com/auth/cloud-platform'])
+    http = httplib2.Http()
+    scoped_credentials.authorize(http)
+    return discovery.build('vision', 'v1',
+                           http=http,
+                           discoveryServiceUrl=DISCOVERY_URL)
+
+
+def crop_hint(photo_path):
+    """Run a crop hint request on the image."""
+
+    service = get_service()
+
+    with open(photo_path, 'rb') as image:
+        image_content = base64.b64encode(image.read())
+        service_request = service.images().annotate(body={
```

Comment:
Your `with` statement covers too many things. It only needs to cover the `image.read` call.